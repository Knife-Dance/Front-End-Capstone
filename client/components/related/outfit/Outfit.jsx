import React, {useState, useEffect, useContext} from 'react';
import css from './outfit.module.css'
import MainContext from '../../shared/context/MainContext.js';
import ReviewAverage from '../../overview/ReviewAverage/ReviewAverage.jsx';





const Outfit = (props) => {
//selected product, handle getStyle by Id, handle get rate by id
//just change the selecte dproduct to the new product
  const [firstItemIndexToShow, setFirstItemIndexToShow] = useState(0);
  const { outfits, addOutfit, removeOutfit, clickListener } = useContext(MainContext);
  // console.log(outfits.slice(0,4))
  const handlePrevClick = () => {
    setFirstItemIndexToShow(firstItemIndexToShow - 1)
  }
  const handleNextClick = () => {
    setFirstItemIndexToShow(firstItemIndexToShow + 1)
  }

  const component = 'Outfit Component'

  return (
    <>
    <h3>YOUR OUTFIT</h3>


    <div className={css.cardContainer} onClick={(e) => clickListener(e, component)}>
                {firstItemIndexToShow !== 0&&
                <span className={css.btnPrev} onClick={handlePrevClick}>
                    <i className="fas fa-chevron-left"></i>
                </span>}


                {firstItemIndexToShow !== outfits.length - 3 &&
                <span className={css.btnNext} onClick={handleNextClick}>
                <i className="fas fa-chevron-right slide"></i>
            </span>
                }
                <div className={css.gridContainer}>

                <div onClick={addOutfit} id={'addOutfit'} className={`${css.addBtn}`} style={{border: '1px solid gray'}}>
                   <h4>Add to Outfit</h4>
                    <i className={'fas fa-plus'}></i>
                 </div>

                    {
                        outfits.slice(firstItemIndexToShow, firstItemIndexToShow + 3).map((item, key) => (
                            <div key={key}
                            style={{border: '1px solid gray'}} className={css.productItem}>

                                <button aria-label="button" id={'outfitCloseBtn'}className= {css.closingBtn}onClick={() => {

                                  removeOutfit(item.style.style_id)
                                  }}>
                                <i className="fas fa-times"></i>
                                </button>

                                {item.style.photos[0].thumbnail_url ? <div  className={css.productImageContainer}><img  alt='outfit img' src={item.style.photos[0].thumbnail_url}/></div> :
                                <div className={css.productImageContainer}>
                                    <img alt='no img found'className={css.noImg} src={noImgFound}/> </div>}
                                    <div className={css.cardInfo}>
                                      <p className={css.category}>{item.product.category}</p>
                                      <p className={css.name}>{item.product.name}</p>

                                      {item.style.sale_price ?  <p className={css.lineThrough}>${item.style.original_price}</p> : <p className={css.nolineThrough}>${item.style.original_price}</p>}

                                      {item.style.sale_price !== null ? <p className={css.salePrice}>{item.style.sale_price}</p> : null}
                                      <ReviewAverage  average={item.rate}/>
                                   </div>


                            </div>
                        ))
                    }

                </div>
            </div>


    </>

  )

}



export default Outfit;


