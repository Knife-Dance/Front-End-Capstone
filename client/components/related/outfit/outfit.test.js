
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import sinon from 'sinon'
import $ from 'jquery';
import Relate from '../product-card/Related-product';
import Outfit from './Outfit';
import css from './outfit.module.css';
import TestProvider from '../../shared/context/TestProvider.jsx'
import {addOutfit} from '../../shared/context/TestProvider.jsx'
import {removeOutfit} from '../../shared/context/TestProvider.jsx'
// console.log(addOutfit)

Enzyme.configure({ adapter: new Adapter() });


describe('Outfit Component', () => {


  it('Renders Modal component', () => {
    const wrapper = shallow( <Outfit/>);;
    expect(wrapper).toMatchSnapshot()
  });

  it('Modal should contain a table', () => {
    const wrapper = shallow( <Outfit/>);
    expect(wrapper.find('button')).toBeTruthy();
  });

  it('Render the Header', () => {
    const wrapper = mount( <Outfit />);
    const btn = wrapper.find('h4').text()
    expect(btn).toEqual('Add to Outfit')
  });

  it('Render button', () => {
    const wrapper = mount( <Outfit />);
    const btn = wrapper.find('i .fa-times')
    expect(btn).toBeTruthy()
  });

  it('Test click event for add button', () => {
   //create an event handler
    // const mockCallBack = jest.fn();
     //render the component

    const button = mount((
      <TestProvider>
        <Outfit />
      </TestProvider>


    ));
        // console.log(button.debug())

    //find the element div
    const btn = button.find('#addOutfit');

    //simulate theclick event on the element
    btn.simulate('click')

      //test to see if simulation goes thru
      expect(addOutfit.mock.calls.length).toBe(1);
  });

  it('Test click event for closing button ', () => {
     const button = mount((
       <TestProvider>
         <Outfit />
       </TestProvider>
     ));

     const btn = button.find('#outfitCloseBtn');

     //simulate theclick event on the element
     btn.simulate('click')

       //test to see if simulation goes thru
       expect(removeOutfit.mock.calls.length).toBe(1);
   });






});
