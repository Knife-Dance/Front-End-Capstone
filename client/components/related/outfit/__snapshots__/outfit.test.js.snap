// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Outfit Component Renders Modal component 1`] = `ShallowWrapper {}`;

exports[`Outfit Component Test click event 1`] = `ReactWrapper {}`;
